status-resolver/src/main/java/com/ackee/versioupdatehandler/UpdateDialog.kt

```diff
@@ -0,0 +1,100 @@
+package com.ackee.versioupdatehandler
+
+import android.app.Dialog
+import android.content.ActivityNotFoundException
+import android.content.Intent
+import android.net.Uri
+import android.os.Bundle
+import androidx.appcompat.app.AlertDialog
+import androidx.fragment.app.DialogFragment
+import com.ackee.versioupdatehandler.model.DialogSettings
+
+/**
+ * Dialog indicating that user should update application.
+ *
+ *
+ * Texts and package name are customizable via [DialogSettings] class
+ */
+class UpdateDialog : DialogFragment() {
+
+    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
```

Comment:
this method is not really nice. Can you please improve it by splitting it to multiple smaller methods?

Reply:
dc69c82a809710daafa827a2c5daad10a0d5f2c2